include/cuco/pair.cuh

```diff
@@ -142,6 +142,18 @@ template <class T1, class T2, class U1, class U2>
 __host__ __device__ constexpr bool operator==(cuco::pair<T1, T2> const& lhs,
                                               cuco::pair<U1, U2> const& rhs) noexcept;
 
+template <typename T>
+struct is_cuco_pair : cuda::std::false_type {};
+
+template <typename F, typename S>
+struct is_cuco_pair<cuco::pair<F, S>> : cuda::std::true_type {};
+
+/// Trait determing whether a given type is tuple-like or not
+template <typename T>
+struct is_tuple_like : cuda::std::disjunction<is_cuco_pair<T>,
+                                              detail::is_std_pair_like<T>,
+                                              detail::is_cuda_std_pair_like<T>> {};
```

Comment:
The other traits are located in `detail/pair/traits.hpp`. Shall we move them?

Reply:
Yeah, that's also the first place I wanted to put this trait but it requires `detail/pair/traits.hpp` to include `cuco/pair.cuh` header otherwise it won't work. The same for `detail/utility/traits.hpp`.

Reply:
Cyclic include. I see. Thanks for the clarification!